kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/SharedKafkaProxyContext.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyspec.Clusters;
+import io.kroxylicious.kubernetes.operator.config.RuntimeDecl;
+
+/**
+ * Encapsulates access to the mutable state in the {@link Context<KafkaProxy>} shared between
+ * {@link ProxyReconciler} its dependent resources.
+ */
+public class SharedKafkaProxyContext {
+
+    private SharedKafkaProxyContext() {
+    }
+
+    static final String RUNTIME_DECL_KEY = "runtime";
+    static final String ERROR_KEY = "error";
+
+    /**
+     * Set the RuntimeDecl
+     */
+    static void runtimeDecl(Context<KafkaProxy> context, RuntimeDecl runtimeDecl) {
+        context.managedDependentResourceContext().put(RUNTIME_DECL_KEY, runtimeDecl);
+    }
+
+    /**
+     * Get the RuntimeDecl
+     */
+    static RuntimeDecl runtimeDecl(Context<KafkaProxy> context) {
+        return context.managedDependentResourceContext().getMandatory(RUNTIME_DECL_KEY, RuntimeDecl.class);
+    }
+
+    /**
+     * Associate an Accepted condition with a specific cluster.
+     */
+    static void addClusterCondition(Context<KafkaProxy> context, Clusters cluster, ClusterCondition clusterCondition) {
+        Map<String, ClusterCondition> map = context.managedDependentResourceContext().get(ERROR_KEY, Map.class).orElse(null);
+        if (map == null) {
+            map = Collections.synchronizedMap(new HashMap<>());
+            context.managedDependentResourceContext().put(ERROR_KEY, map);
+        }
+        map.put(cluster.getName(), clusterCondition);
+    }
+
+    static boolean isBroken(Context<KafkaProxy> context, Clusters cluster) {
+        Map<String, ClusterCondition> map = context.managedDependentResourceContext().get(ERROR_KEY, Map.class).orElse(Map.of());
+        return map.containsKey(cluster.getName());
+    }
```

Comment:
Did you change strategies at some point? As I didn't think `acceptedConditons` represented errors.

Reply:
That's correct. If you read the code closely you'll see that accepted condition never and up in the map. So I think this code is still good. 

It's worth pointing out that this `SharedKafkaProxyContext` is a tricky thing. That state is mutable and is propagated between the dependent resources and the reconciler that gets to set the status. So it's possible that the first `DependentResources` don't see errors, but eventually one reports an error and so subsequent ones, and the reconciler, will see those errors. I don't currently see a great way to avoid this within the annotation-based declarative model of JOSDK. I don't see it as a huge problem right now, given we have just three `DependentResources`. It will become more of a problem if that number grows. But again, I don't want to do something more elaborate just yet as least until I've done the `KafkaRoute` refactor and we understand how having two reconcilers alters the picture.